core/src/main/scala/io/gearpump/transport/netty/Client.scala

```diff
@@ -200,11 +200,11 @@ object Client {
     }
   }
 
-  class ClientPipelineFactory(name: String, decoder: MessageDecoder, encoder: MessageEncoder) extends ChannelPipelineFactory {
+  class ClientPipelineFactory(name: String, conf: NettyConfig) extends ChannelPipelineFactory {
```

Comment:
I think the original interface is better for testing where decoder and encoder could be mocked. Same holds for ServerPipelineFactory